src/js/components/user/connect.tsx

```diff
@@ -1,18 +1,153 @@
-import React from 'react';
+import Button from '@salesforce/design-system-react/components/button';
+import Input from '@salesforce/design-system-react/components/input';
+import Modal from '@salesforce/design-system-react/components/modal';
+import i18n from 'i18next';
+import React, { useState } from 'react';
+import { Trans } from 'react-i18next';
 
-import Login from './sfLogin';
+import { ExternalLink } from '@/components/utils';
+import { addUrlParams } from '@/utils/api';
 
-export interface Props {}
+const CustomDomainForm = ({
+  url,
+  setUrl,
+}: {
+  url: string;
+  setUrl: React.Dispatch<React.SetStateAction<string>>;
+}) => {
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const val = url.trim();
+    if (!val) {
+      return;
+    }
+    const baseUrl = window.api_urls.salesforce_custom_login();
+    window.location.assign(
+      addUrlParams(baseUrl, {
+        custom_domain: val, // eslint-disable-line @typescript-eslint/camelcase
+        next: window.location.pathname,
+      }),
```

Comment:
Somewhere around here you should have `process: 'connect'`. But also for the production endpoint, not just the custom endpoint.